Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperations/TestOperation.swift

```diff
@@ -0,0 +1,298 @@
+import MongoSwiftSync
+import Nimble
+import TestsCommon
+
+/// Protocol describing the behavior of a spec test "operation"
+protocol TestOperation: Decodable {
+    func execute(on client: MongoClient, sessions: [String: ClientSession]) throws -> TestOperationResult?
+
+    func execute(on database: MongoDatabase, sessions: [String: ClientSession]) throws -> TestOperationResult?
+
+    func execute(
+        on collection: MongoCollection<Document>,
+        sessions: [String: ClientSession]
+    ) throws -> TestOperationResult?
+
+    func execute(on session: ClientSession) throws -> TestOperationResult?
+
+    func execute<T: SpecTest>(on runner: inout T, sessions: [String: ClientSession]) throws -> TestOperationResult?
+
+    /// The name of the session this operation should execute against, if any.
+    var session: String? { get }
```

Comment:
this was added so that the `getSession` method in the extension below could be implemented.